src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/SecurityDynamicConfiguration.java

```diff
@@ -48,18 +49,57 @@
 import com.fasterxml.jackson.annotation.JsonAnyGetter;
 import com.fasterxml.jackson.annotation.JsonAnySetter;
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JavaType;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.introspect.BeanPropertyDefinition;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.BeanPropertyWriter;
+import com.fasterxml.jackson.databind.ser.BeanSerializer;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+
 import com.amazon.opendistroforelasticsearch.security.DefaultObjectMapper;
 import com.amazon.opendistroforelasticsearch.security.NonValidatingObjectMapper;
 import com.amazon.opendistroforelasticsearch.security.securityconf.Hashed;
 import com.amazon.opendistroforelasticsearch.security.securityconf.Hideable;
 import com.amazon.opendistroforelasticsearch.security.securityconf.StaticDefinable;
-import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v6.ConfigV6;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.ConfigV7;
 
 public class SecurityDynamicConfiguration<T> implements ToXContent {
     
     private static final TypeReference<HashMap<String,Object>> typeRefMSO = new TypeReference<HashMap<String,Object>>() {};
+    private static final String FOR_DISPLAY = "for_display";
+    public static final ToXContent.MapParams FOR_DISPLAY_PARAMS = new ToXContent.MapParams(Collections.singletonMap(FOR_DISPLAY, "true"));
```

Comment:
Could you add more information about `for_display` in PR/commit description ? Cannot see much information around this and what it means.

Reply:
`FOR_DISPLAY_PARAMS` changes `objectMapper` that is used to serialize `SecurityDynamicConfiguration` to JSON to include properties that will not be serialized by default `objectMapper` when they are `null` or empty.

Reply:
Is "for_display" supposed to mean something?

Reply:
Yes, it means not for persistence. I can rename it to `TRANSIENT_PARAMS`. It is used to generate GET response in REST API.